pvnet/models/multimodal/validation/config_batch_validation.py

```diff
@@ -0,0 +1,194 @@
+"""Validation functoions for Multimodal batches."""
+
+import logging
+from typing import Any, Optional, Type
+
+import numpy as np
+from ocf_data_sampler.torch_datasets.sample.base import NumpyBatch
+
+logger = logging.getLogger(__name__)
+
+
+def check_batch_data(
+    numpy_batch: NumpyBatch,
+    batch_key: str,
+    expected_type: Type,
+    context_config_key: str,
+) -> Any:
+    """Check if modality data exists in batch and has the expected type."""
+    if batch_key not in numpy_batch:
+        raise KeyError(
+            f"Batch missing required '{batch_key}' data "
+            f"(required by config key '{context_config_key}')."
+        )
+    data: Any = numpy_batch[batch_key]
+    if not isinstance(data, expected_type):
+        expected_type_name = getattr(expected_type, '__name__', str(expected_type))
+        raise TypeError(
+            f"'{batch_key}' data must be {expected_type_name}, "
+            f"found {type(data).__name__}"
+        )
+    return data
+
+
+def get_modality_interval(
+    input_data_config: dict,
+    primary_modality_key: str,
+    secondary_modality_key: Optional[str] = None,
```

Comment:
```suggestion
    secondary_modality_key: str | None,
```

I think we generally try to stick to the newer union syntax elswere in the codebase (or at least in dsampler)